api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/art/NCDStatusDataEvaluator.java

```diff
@@ -34,13 +34,19 @@ public class NCDStatusDataEvaluator implements PersonDataEvaluator {
     public EvaluatedPersonData evaluate(PersonDataDefinition definition, EvaluationContext context) throws EvaluationException {
         EvaluatedPersonData c = new EvaluatedPersonData(definition, context);
 
-        String qry = "select ci.patient_id,\n" +
-                "       group_concat(case ci.is_chronic_illness_controlled\n" +
-                "                        when 1065 then 'Controlled'\n" +
-                "                        when 1066 then 'Not controlled' end) as controlled_status\n" +
-                "from kenyaemr_etl.etl_allergy_chronic_illness ci\n" +
-                "where ci.visit_date <= date(:endDate)\n" +
-                "group by ci.patient_id;";
+        String qry = "SELECT ci.patient_id,\n" +
+                "       (CASE ci.is_chronic_illness_controlled\n" +
```

Comment:
You need to handle a case where there are more than on NCD recorded in that visit

Reply:
Resolved the changes. Thanks